src/coord/axisTickLabelBuilder.ts

```diff
@@ -93,12 +112,32 @@ export function createAxisTicks(axis: Axis, tickModel: AxisBaseModel): {
         : {ticks: zrUtil.map(axis.scale.getTicks(), tick => tick.value) };
 }
 
+/**
+ * @param {module:echats/coord/Axis} axis
+ *
+ */
+export function getAxisNameGap(axis: Axis): number {
+    const axesModel = axis.model;
+    const nameGap = axesModel.get('nameGap') ?? 0;
+    if (axesModel.get('nameLayout') === 'auto'
+        && isNameLocationCenter(axis.model.get('nameLocation'))
+    ) {
+        const labelUnionRect = estimateLabelUnionRect(axis);
```

Comment:
For centered labels with `nameLayout: 'auto'`, calculate the name gap dynamically to "stack" the axis name together with the labels:

<table><tr>
<td><img width="651" alt="Screenshot 2023-11-30 at 1 27 33 PM" src="https://github.com/apache/echarts/assets/3394202/637b1452-4dbd-4dd8-8ee5-8be937d605ee"></td>
<td><img width="343" alt="Screenshot 2023-11-30 at 1 28 00 PM" src="https://github.com/apache/echarts/assets/3394202/ec501fd0-ac65-48b2-a921-bc48e8af7742"></td>
<td><img width="283" alt="Screenshot 2023-11-30 at 1 28 15 PM" src="https://github.com/apache/echarts/assets/3394202/16f02a7d-ca08-44e7-ab9f-f2fbbdc3a5f7"></td>
<td><img width="258" alt="Screenshot 2023-11-30 at 1 28 43 PM" src="https://github.com/apache/echarts/assets/3394202/1a1fe6d5-97e9-40f7-9428-ba0df7efc879"></td>
</tr></table>